src/mousetracker.js

```diff
@@ -2623,60 +3419,153 @@
      * @inner
      * @param {OpenSeadragon.MouseTracker} tracker
      *     A reference to the MouseTracker instance.
-     * @param {Object} event
-     *     A reference to the originating DOM event.
-     * @param {Array.<OpenSeadragon.MouseTracker.GesturePoint>} gPoints
-     *      Gesture points associated with the event.
+     * @param {OpenSeadragon.MouseTracker.EventProcessInfo} eventInfo
+     *     Processing info for originating DOM event.
+     * @param {OpenSeadragon.MouseTracker.GesturePoint} gPoint
+     *      Gesture point associated with the event.
      */
-    function updatePointersExit( tracker, event, gPoints ) {
-        var pointsList = tracker.getActivePointersListByType(gPoints[0].type),
-            i,
-            gPointCount = gPoints.length,
-            curGPoint,
+    function updatePointerLeave( tracker, eventInfo, gPoint ) {
+        var pointsList = tracker.getActivePointersListByType(gPoint.type),
             updateGPoint,
-            propagate;
-
-        for ( i = 0; i < gPointCount; i++ ) {
-            curGPoint = gPoints[ i ];
-            updateGPoint = pointsList.getById( curGPoint.id );
-
-            if ( updateGPoint ) {
-                // Already tracking the pointer. If captured then update it, else stop tracking it
-                if ( updateGPoint.captured ) {
-                    updateGPoint.insideElement = false;
-                    updateGPoint.lastPos = updateGPoint.currentPos;
-                    updateGPoint.lastTime = updateGPoint.currentTime;
-                    updateGPoint.currentPos = curGPoint.currentPos;
-                    updateGPoint.currentTime = curGPoint.currentTime;
-                } else {
-                    stopTrackingPointer( pointsList, updateGPoint );
-                }
+            dispatchEventObj;
 
-                curGPoint = updateGPoint;
+        updateGPoint = pointsList.getById( gPoint.id );
+
+        if ( updateGPoint ) {
+            // Already tracking the pointer. If captured then update it, else stop tracking it
+            if ( updateGPoint.captured ) {
+                updateGPoint.insideElement = false;
+                updateGPoint.lastPos = updateGPoint.currentPos;
+                updateGPoint.lastTime = updateGPoint.currentTime;
+                updateGPoint.currentPos = gPoint.currentPos;
+                updateGPoint.currentTime = gPoint.currentTime;
+            } else {
+                stopTrackingPointer( tracker, pointsList, updateGPoint );
             }
 
-            // Exit
+            gPoint = updateGPoint;
+        } else {
+            gPoint.captured = false; // Handled by updatePointerCaptured()
+            gPoint.insideElementPressed = false;
+        }
+
+        // Leave (doesn't bubble and not cancelable)
+        //   Note: exitHandler is deprecated, replaced by leaveHandler
+        if ( tracker.leaveHandler || tracker.exitHandler ) {
```

Comment:
@iangilman  Here's where the deprecated exitHandler is handled. Pondering if I should use an else-if so only one handler is called...and maybe emit a warning to the console if it's exitHandler?

Reply:
Thank you for pointing it out here! I think it's probably fine if both handlers are called (typically if you have an exit handler you wouldn't have a leave handler anyway). I think it would be good to console.warn if it's the exit handler, though.